xoa_driver/functions/anlt.py

```diff
@@ -20,191 +28,279 @@
+                        timeout_mode=TimeoutMode.DEFAULT
+                    ),
+                    commands.PP_LINKTRAIN(conn, mid, pid).set(
+                    mode=LinkTrainingMode.INTERACTIVE,
+                    pam4_frame_size=PAM4FrameSize.P16K_FRAME,
+                    nrz_pam4_init_cond=LinkTrainingInitCondition.NO_INIT,
+                    nrz_preset=NRZPreset.NRZ_WITH_PRESET if lt_preset0_std else NRZPreset.NRZ_NO_PRESET,
+                    timeout_mode=TimeoutMode.DISABLED
+                    )
+                ]
+            else:
+                # LT only (standalone mode)
+                tokens += [
+                    # Disable autoneg
+                    commands.PP_AUTONEG(conn, mid, pid).set(
+                        AutoNegMode.ANEG_OFF,
+                        AutoNegTecAbility.DEFAULT_TECH_MODE,
+                        AutoNegFECOption.NO_FEC,
+                        AutoNegFECOption.NO_FEC,
+                        PauseMode.NO_PAUSE,
+                    ),
+                ]
+                # Configure link training initial modulations for the specified lanes (serdes)
+                for lane_str, im in lt_initial_modulations.items():
+                    tokens += [
+                        commands.PL1_CFG_TMP(conn, mid, pid, int(lane_str), Layer1ConfigType.LT_INITIAL_MODULATION).set(
+                            values=[im]
+                        )
+                    ]
+                tokens += [
+                    # Set link training state back to DISABLE first and then STANDALONE
+                    commands.PP_LINKTRAIN(conn, mid, pid).set(
+                        mode=LinkTrainingMode.DISABLED,
+                        pam4_frame_size=PAM4FrameSize.P16K_FRAME,
+                        nrz_pam4_init_cond=LinkTrainingInitCondition.NO_INIT,
+                        nrz_preset=NRZPreset.NRZ_WITH_PRESET if lt_preset0_std else NRZPreset.NRZ_NO_PRESET,
+                        timeout_mode=TimeoutMode.DEFAULT
+                    ),
+                    commands.PP_LINKTRAIN(conn, mid, pid).set(
+                    mode=LinkTrainingMode.STANDALONE,
+                    pam4_frame_size=PAM4FrameSize.P16K_FRAME,
+                    nrz_pam4_init_cond=LinkTrainingInitCondition.NO_INIT,
+                    nrz_preset=NRZPreset.NRZ_WITH_PRESET if lt_preset0_std else NRZPreset.NRZ_NO_PRESET,
+                    timeout_mode=TimeoutMode.DISABLED
+                    )
+                ]
+        else:
+            # No AN, no LT
+            tokens += [
+                commands.PP_AUTONEG(conn, mid, pid).set(
+                    AutoNegMode.ANEG_OFF,
+                    AutoNegTecAbility.DEFAULT_TECH_MODE,
+                    AutoNegFECOption.NO_FEC,
+                    AutoNegFECOption.NO_FEC,
+                    PauseMode.NO_PAUSE,
+                )
+            ]
+        await apply(*tokens)
```

Comment:
WTF with this code, please improve this function, by split actions, and avoiding of unnecessary aggregation.
>## Function contain a bug, the `apply` method is not properly placed, which will cost to not all logical forks will work. this issue is difficult to trace coz the function is having to many code indentation levels